### PR TITLE
Document MFA workaround

### PR DESCRIPTION
**Pull request by ahawkins**
I figure we can throw this into the docs so end users have a documented work around until there is real support. Please let me know if there is a better way to document this (or an existing doc to add to).

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1888)

### CODE REVIEW

File: docs/mfa.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+set -euo pipefail
+
+main() {
+	local role_arn="${KOPS_MFA_ROLE_ARN:-}"
+	local serial_number="${KOPS_MFA_ARN:-}"
+	local token_code
+
+	if [ -z "${KOPS_MFA_ROLE_ARN}" ]; then
+		echo "Set the KOP_MFA_ROLE_ARN environment variable" 1>&2;
```
Comment (justinsb) [inline docs/mfa.md]: typo: should be `KOPS_MFA_ROLE_ARN`

File: docs/mfa.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# kops & MFA
+
+You can secure `kops` with MFA by creating an AWS role & policy that requires MFA to access to the `KOPS_STATE_STORE` bucket. Unfortunately the Go AWS SDK does not transparently support assuming roles with required MFA. This may change in a future version. `kops` plans to support this behavior eventually. You can track progress in this [Github issue](https://github.com/kubernetes/kops/issues/226). If you'd like to ues MFA with `kops`, you'll need a work around until then.
+
+## The Workaround
+
+The work around uses `aws sts-assume role` in combination with an MFA prompt to retrieve temporary AWS access keys. This provides `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_SESSSION` token environment variables which are automatically picked up by Go AWS SDK. You provide the MFA & Role ARNs, then invoke `kops`.
```
Comment (justinsb) [inline docs/mfa.md]: Typo: `AWS_SESSION`

**Comment by ahawkins** [comment]
@justinsb how about now?

**Comment by kris-nova** [comment]
@ahawkins there is still a typo on line 7

> AWS_SESSSION_TOKEN

should be

> AWS_SESSION_TOKEN

**Comment by kris-nova** [comment]
Can you fix the typo and squash all 4 commits together as well?

**Comment by ahawkins** [comment]
Should be good to go now!

**Comment by kris-nova** [comment]
Hrmm it looks like `make ci` is failing:

```
FAIL	k8s.io/kops/cmd/kops	10.571s
```

Albeit your changes are documentation only, so I am wondering if something more sinister is at play. @justinsb any ideas?

I can take a look later on today if we haven't heard anything.

**Comment by justinsb** [comment]
Going to close & reopen to try again with travis.

I think travis builds the PR branch, not the merged PR.  So if head was broken, I don't think travis "rebases".

If this fails after a close/reopen, I'll just merge - if it breaks head I'll fix it - but I don't think it will!

**Comment by chrislovecnm** [comment]
@k8s-bot ok to test

**Comment by ahawkins** [comment]
@chrislovecnm is this PR still valid? I saw your comment in another issue. If so, how can we move this one along?

**Pull request merged**